src/urllib3/connection.py

```diff
@@ -410,6 +401,10 @@ def connect(self):
         )
 
 
+class VerifiedHTTPSConnection(HTTPSConnection):
+    pass
+
```

Comment:
I think that on the lines 423-428:
```python
if ssl:
    # Make a copy for testing.
    UnverifiedHTTPSConnection = HTTPSConnection
    HTTPSConnection = VerifiedHTTPSConnection
else:
    HTTPSConnection = DummyConnection
```

We should consider  the following:
- The `HTTPSConnection` class should not be shadowed by reassignment so that it is exported as is (im not sure about the use case of the `DummyConnection` in the else statement)
- if we are keeping `UnverifiedHTTPSConnection = HTTPSConnection` then we should also have `VerifiedHTTPSConnection = HTTPSConnection` and then remove `VerifiedHTTPSConnection` class definition code.
  -  Alternatively, a `UnverifiedHTTPSConnection` class might need to be defined just like how `VerifiedHTTPSConnection` is defined.

Reply:
Thanks for your review! Overlooked these lines. Could we delete the if statement? I like the third option you mentioned, so defining `UnverifiedHTTPSConnection` like `VerifiedHTTPSConnection` is simpler IMO. 
I am not sure the purpose of `DummyConnection`.
Let me know any thoughts.

Reply:
`DummyConnection` seems to be used only for here.
https://github.com/urllib3/urllib3/blob/75d271954b5d662d0567e95b054475e208619dff/src/urllib3/connectionpool.py#L962-L965

Reply:
I'm thinking we do something like:

```
if ssl:
    HTTPSConnection = DummyConnection

VerifiedHTTPSConnection = HTTPSConnection
```

so we still get the nice error if the user is running Python without an `ssl` module and always get the name rebound to `VerifiedHTTPSConnection`. :)

Since the `UnverifiedHTTPSConnection` class was totally borked for the past ~6 months and we only received one issue on it I think we're good to remove it in the next release. :tada: Just need to document in the changelog that there's only one HTTPSConnection object now. :)

Reply:
Thanks @sethmlarson, Let me check the if statement. We assign `DummyConnection` to `HTTPSConnection` if not ssl, right? So the if statement is like this?

```
if not ssl:
    HTTPSConnection = DummyConnection

VerifiedHTTPSConnection = HTTPSConnection
```

Reply:
You're absolutely right haha, nice catch. :)

Reply:
Thank you! modified the statement and added to CHANGES.rst